libsolidity/analysis/TypeChecker.cpp

```diff
@@ -743,7 +743,7 @@ bool TypeChecker::visit(InlineAssembly const& _inlineAssembly)
 			solAssert(var->type(), "Expected variable type!");
 			if (var->immutable())
 			{
-				m_errorReporter.typeError(3773_error, _identifier.location, "Assembly access to immutable variables is not supported.");
+				m_errorReporter.typeError(3773_error, _identifier.debugData->irLocation, "Assembly access to immutable variables is not supported.");
```

Comment:
Thinking a bit more about this, I'm not 100% sold that the `visit(InlineAssembly)` section actually wants to use the `irLocation` and not the yul location.

The irLocation comes into play when created source mappings at the end of the IR pipeline.